controllers/datadogagent/override/global.go

```diff
@@ -199,16 +215,25 @@ func ApplyGlobalSettings(logger logr.Logger, manager feature.PodTemplateManagers
 					agentCAPath = apicommon.KubeletAgentCAPath
 				}
 				kubeletVol, kubeletVolMount := volume.GetVolumes(apicommon.KubeletCAVolumeName, config.Kubelet.HostCAPath, agentCAPath, true)
-				manager.VolumeMount().AddVolumeMountToContainers(
-					&kubeletVolMount,
-					[]apicommonv1.AgentContainerName{
-						apicommonv1.CoreAgentContainerName,
-						apicommonv1.ProcessAgentContainerName,
-						apicommonv1.TraceAgentContainerName,
-						apicommonv1.SecurityAgentContainerName,
-					},
-				)
-				manager.Volume().AddVolume(&kubeletVol)
+				if usesMultiProcessCoreAgent {
+					manager.VolumeMount().AddVolumeMountToContainers(
+						&kubeletVolMount,
+						[]apicommonv1.AgentContainerName{
+							apicommonv1.UnprivilegedMultiProcessAgentContainerName,
+						},
+					)
+					manager.Volume().AddVolume(&kubeletVol)
+				} else {
+					manager.VolumeMount().AddVolumeMountToContainers(
+						&kubeletVolMount,
+						[]apicommonv1.AgentContainerName{
+							apicommonv1.CoreAgentContainerName,
+							apicommonv1.ProcessAgentContainerName,
+							apicommonv1.TraceAgentContainerName,
+						},
```

Comment:
is the `apicommonv1.SecurityAgentContainerName` missing here?

Reply:
oh, thanks for catching. It looks this was added after PR was opened and probably lost during rebase/conflict resolution https://github.com/DataDog/datadog-operator/pull/950/files. Fixed now.